src/HPWH.cc

```diff
@@ -1207,22 +1227,108 @@ void HPWH::updateTankTemps(double drawVolume_L, double inletT_C, double tankAmbi
 	} //end if(draw_volume_L > 0)
 
 
-	//calculate standby losses
-	//get average tank temperature
-	double avgTemp = 0;
-	for (int i = 0; i < numNodes; i++) avgTemp += tankTemps_C[i];
-	avgTemp /= numNodes;
+	// calculate conduction between the nodes AND heat loss by node with top and bottom having greater surface area.
+	// model uses explicit finite difference to find conductive heat exchange between the tank nodes with the boundary conditions
+	// on the top and bottom node being the fraction of UA that corresponds to the top and bottom of the tank.  
+	if (doConduction) {
+		// height estimate from Rheem along with the volume is used to get the radius and node_height
+		double height = 1.2; //meters
+		double rad = sqrt(tankVolume_L / (1000 * 3.14159 * height));
+		double node_height = height/numNodes;
+
+		// Get the "constant" tau for the stability condition and the conduction calculation
+		double tau = KWATER_WpermC / ( CPWATER_kJperkgC * 1000 * DENSITYWATER_kgperL * 1000 * (node_height * node_height) ) * minutesPerStep * 60.0;
+		if (tau > 0.5) msg("The stability condition for conduction has failed, these results are going to be interesting!\n");
+
+		// The fraction of UA that is on the top and bottom of the tank
+		double UA_bt = tankUA_kJperHrC * rad / (2 * (height + rad));
+		double bc = 2 * tau * UA_bt * node_height / KWATER_WpermC;
+
+		// Boundary nodes for finite difference
+		tempTemps[0] = (1 - 2 * tau - bc) * tankTemps_C[0] + 2 * tau * tankTemps_C[1] + bc * tankAmbientT_C;
+		tempTemps[numNodes - 1] = (1 - 2 * tau - bc) * tankTemps_C[numNodes - 1] + 2 * tau * tankTemps_C[numNodes - 2] + bc * tankAmbientT_C;
+
+		// Internal nodes for the finite difference
+		for (int i = 1; i < numNodes - 1; i++) {
+			tempTemps[i] = tankTemps_C[i] + tau * (tankTemps_C[i + 1] - 2 * tankTemps_C[i] + tankTemps_C[i - 1]);
+		}
+		// tempTemps gets assigns to tankTemps_C at the bottom of the function after q_UA.
+		
+		// Ar is the faction of the area of the cylinder that's not the top or bottom.
+		double Ar = height / (height + rad);
+
+		//calculate standby losses
+		for (int i = 0; i < numNodes; i++) {
+			//faction of tank area on the sides
+			//kJ's lost as standby in the current time step for each node.
+			double standbyLosses_kJ = (tankUA_kJperHrC * Ar / numNodes * (tankTemps_C[i] - tankAmbientT_C) * (minutesPerStep / 60.0));
+			standbyLosses_kWh += KJ_TO_KWH(standbyLosses_kJ);
+
+			//The effect of standby loss on temperature in each node
+			tempTemps[i] -= standbyLosses_kJ / ((volPerNode_LperNode * DENSITYWATER_kgperL) * CPWATER_kJperkgC);
+		}
+	}
+	else { // Ignore tank conduction and calculate UA losses in the old way.

```

Comment:
So I changed it, but it's pretty screwy. Because the top node loses heat faster than the other nodes it constantly creates inversions. I made two plots below, the first is temperature, note the green line is the top node, and the second is the output power and the hot water draws. This also seems to result in weird behavior for the compressor.

I'm fine leaving this in as a setting for the research build but just thought you should be warned.  

![image](https://user-images.githubusercontent.com/50156325/63392790-fcff8800-c36c-11e9-846a-ea1f275d355a.png)

![image](https://user-images.githubusercontent.com/50156325/63392792-fffa7880-c36c-11e9-9826-6489051dc2c3.png)

Reply:
It seems like we should always do inversion mixing, too.